addons/knowledge/static/src/js/widgets/knowledge_dialogs.js

```diff
@@ -0,0 +1,186 @@
+                },
+                /**
+                 * @param {Object} params - parameters
+                 */
+                transport: async params => {
+                    const { term } = params.data;
+                    const { state } = this.props;
+                    const results = await this._rpc({
+                        model: 'knowledge.article',
+                        method: 'get_possible_parents',
+                        args: [state.data.id],
+                        kwargs: {
+                            term
+                        }
+                    });
+                    params.success({ term, results });
+                },
+                /**
+                 * @param {Object} data
+                 * @returns {Object}
+                 */
+                processResults: function (data) {
+                    const records = { results: [] };
+                    for (const result of cache.results) {
+                        if (typeof result.children === 'undefined') {
+                            records.results.push(result);
+                            continue;
+                        }
+                        const children = result.children.filter(child => {
+                            const text = child.text.toLowerCase();
+                            const term = data.term.toLowerCase();
+                            return text.indexOf(term) >= 0;
+                        });
+                        if (children.length > 0) {
+                            records.results.push({...result, children});
+                        }
+                    }
+                    if (data.results.length > 0) {
+                        records.results.push({
+                            text: _t('Articles'),
+                            children: data.results.map(record => {
+                                return {
+                                    id: record.id,
+                                    icon: record.icon,
+                                    text: record.name,
+                                }
+                            })
+                        });
+                    }
+                    return records;
+                },
+            },
+            /**
+             * @param {Object} data
+             * @param {JQuery} container
+             * @param {Function} escapeMarkup
+             */
+            formatSelection: (data, container, escapeMarkup) => {
+                const markup = [];
+                if (data.id === 'private') {
+                    const src = escapeMarkup(this.getLoggedUserPicture());
+                    markup.push(`<img src="${src}" class="rounded-circle mr-1"/>`);
+                } else if (typeof data.icon !== 'undefined') {
+                    markup.push(`<i class="fa fa-w ${escapeMarkup(data.icon)} mr-1"/>`);
+                }
+                markup.push(escapeMarkup(data.text));
+                return markup.join('');
```

Comment:
Could we use our own markup from `const {Markup} = require('web.utils');` instead of this "escapeMarkup"?